src-test/src/com/etendoerp/etendorx/events/CheckEntityInEntityMappingEventHandlerTest.java

```diff
@@ -0,0 +1,184 @@
+    try (MockedStatic<ModelProvider> mockedProvider = mockStatic(ModelProvider.class)) {
+      EntityNewEvent event = mock(EntityNewEvent.class);
+      ModelProvider providerInstance = mock(ModelProvider.class);
+
+      when(entityMapping.getMappingEntity()).thenReturn(TestUtils.VALID_ENTITY);
+      when(event.getTargetInstance()).thenReturn(entityMapping);
+      mockedProvider.when(ModelProvider::getInstance).thenReturn(providerInstance);
+      when(providerInstance.getEntity(eq(TestUtils.VALID_ENTITY), eq(false))).thenReturn(mockEntity);
+
+      handler = spy(new TestableCheckEntityInEntityMappingEventHandler());
+      doReturn(true).when(handler).isValidEvent(any(EntityNewEvent.class));
+
+      handler.onSave(event);
+    }
+  }
+
+  /**
+   * Tests that onUpdate with an invalid entity throws an IllegalArgumentException.
+   */
+  @Test(expected = IllegalArgumentException.class)
+  public void testOnUpdateWithInvalidEntityShouldThrowException() {
+    try (MockedStatic<ModelProvider> mockedProvider = mockStatic(ModelProvider.class);
+         MockedStatic<OBMessageUtils> mockedUtils = mockStatic(OBMessageUtils.class)) {
+
+      EntityUpdateEvent event = mock(EntityUpdateEvent.class);
+      ModelProvider providerInstance = mock(ModelProvider.class);
+
+      when(entityMapping.getMappingEntity()).thenReturn(TestUtils.INVALID_ENTITY);
+      when(event.getTargetInstance()).thenReturn(entityMapping);
+      mockedProvider.when(ModelProvider::getInstance).thenReturn(providerInstance);
+      when(providerInstance.getEntity(eq(TestUtils.INVALID_ENTITY), eq(false))).thenReturn(null);
+
+      mockedUtils.when(() -> OBMessageUtils.getI18NMessage(
+              eq("ETRX_WrongEntityName"),
+              any(String[].class)))
+          .thenReturn("Entity not found");
+
+      handler = spy(new TestableCheckEntityInEntityMappingEventHandler());
+      doReturn(true).when(handler).isValidEvent(any(EntityUpdateEvent.class));
+
+      handler.onUpdate(event);
+    }
+  }
+
+  /**
+   * Tests that onSave with an invalid entity throws an IllegalArgumentException.
+   */
+  @Test(expected = IllegalArgumentException.class)
+  public void testOnSaveWithInvalidEntityShouldThrowException() {
+    try (MockedStatic<ModelProvider> mockedProvider = mockStatic(ModelProvider.class);
+         MockedStatic<OBMessageUtils> mockedUtils = mockStatic(OBMessageUtils.class)) {
+
+      EntityNewEvent event = mock(EntityNewEvent.class);
+      ModelProvider providerInstance = mock(ModelProvider.class);
+
+      when(entityMapping.getMappingEntity()).thenReturn(TestUtils.INVALID_ENTITY);
+      when(event.getTargetInstance()).thenReturn(entityMapping);
+      mockedProvider.when(ModelProvider::getInstance).thenReturn(providerInstance);
+      when(providerInstance.getEntity(eq(TestUtils.INVALID_ENTITY), eq(false))).thenReturn(null);
+
+      mockedUtils.when(() -> OBMessageUtils.getI18NMessage(
+              eq("ETRX_WrongEntityName"),
```

Comment:
`ETRX_WrongEntityName `
I recommend passing to utils as they are used several times